### PR TITLE
Create page for creating route (trip - previous name)

### PR DESCRIPTION
**Pull request by marynakernychna**

### CODE REVIEW

**Review by AndriiChepeliuk — APPROVED** [review]

**Review by AndriiChepeliuk — APPROVED** [review]

File: src/components/createRoute/index.jsx [hunk 1]
```diff
@@ -0,0 +1,813 @@
...
+
+    const selectDestinationAddressAsync = async (destinationAddress) => {
+        const coordinates = await getCoordinatesFromAddress(destinationAddress);
+
+        setDestinationCoordinates(coordinates);
+        setCenter(coordinates);
+        setDestinationAddress(destinationAddress);
+    };
+
+    const selectSubPointAddress = (subPointAddress) => {
```
Comment (Grasmus) [inline src/components/createRoute/index.jsx]: Do you realy need this method? You can directly use setSubPointAddress() method

File: src/components/createRoute/index.jsx [hunk 2]
```diff
@@ -0,0 +1,813 @@
...
+    const [subPointAddress, setSubPointAddress] = useState();
+    const [subPointsAddresses, setSubPointsAddresses] = useState([]);
+
+    // coordinates
+    const [originCoordinates, setOriginCoordinates] = useState([]);
+    const [destinationCoordinates, setDestinationCoordinates] = useState([]);
+    const [subPointCoordinates, setSubPointCoordinates] = useState([]);
+
+    // map
+    const [map, setMap] = useState();
```
Comment (Grasmus) [inline src/components/createRoute/index.jsx]: It is unused valiable, so it can be removed

File: src/components/createRoute/index.jsx [hunk 3]
```diff
@@ -0,0 +1,813 @@
...
+                    distance += parseFloat(legDistanceInM);
+                }
+                else {
+                    legDistanceInM = legDistanceInM.replace(',', '.');
+                    distance += legDistanceInM * 1000;
+                }
+
+                const currentDistanceInKm = distance < 1000 ? 0 : parseFloat(distance / 1000);
+
+                if (currentDistanceInKm - 5 > previousKmPoint) {
```
Comment (Grasmus) [inline src/components/createRoute/index.jsx]: Magic number

File: src/components/createRoute/index.jsx [hunk 4]
```diff
@@ -0,0 +1,813 @@
...
+                    <Form.Item
+                        label="Enter the max route deviation: "
+                        name="maxRouteDeviationKm"
+                        rules={[
+                            InputRules.required(carsErrorMessages.EMPTY_FIELD)
+                        ]}
+                    >
+                        <InputNumber
+                            min={1}
+                            max={25}
```
Comment (Grasmus) [inline src/components/createRoute/index.jsx]: Magic number

File: src/components/createRoute/index.jsx [hunk 5]
```diff
@@ -0,0 +1,813 @@
...
+                    </Form.Item>
+
+                    <Form.Item
+                        label="Enter the description: "
+                        name="description"
+                        rules={[
+                            InputRules.required(
+                                inputValidationErrorMessages.EMPTY_DESCRIPTION
+                            ),
+                            InputRules.lengthRange(
```
Comment (Grasmus) [inline src/components/createRoute/index.jsx:781]: I still can make a description with spaces only

File: src/components/createRoute/index.jsx [hunk 6]
```diff
@@ -0,0 +1,813 @@
...
+                        </PlacesAutocomplete>
+                    </Form.Item>
+
+                    <div className="button">
+                        <Button
+                            onClick={() => addSubPointAsync()}
+                        >
+                            Add sub point
+                        </Button>
+                    </div>
```
Comment (Grasmus) [inline src/components/createRoute/index.jsx:695]: I think it is better to put here Build the route button

File: src/components/createRoute/index.jsx [hunk 7]
```diff
@@ -0,0 +1,813 @@
...
+                                                    className="description-item"
+                                                >
+                                                    {suggestion.description}
+                                                </div>
+                                            );
+                                        })}
+                                    </div>
+                                </div>
+                            )}
+                        </PlacesAutocomplete>
```
Comment (Grasmus) [inline src/components/createRoute/index.jsx:599]: I think you can create a separate component with PlacesAutocomplete and others components within it

**Review by Grasmus — APPROVED** [review]

**Review by AndriiChepeliuk — APPROVED** [review]

**Pull request merged**